src/main/java/org/folio/circulation/resources/ChangeStatusResource.java

```diff
@@ -0,0 +1,86 @@
+package org.folio.circulation.resources;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.domain.validation.LoanValidator.refuseWhenLoanIsClosed;
+import static org.folio.circulation.support.ItemRepository.noLocationMaterialTypeAndLoanTypeInstance;
+import static org.folio.circulation.support.Result.succeeded;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import org.folio.circulation.StoreLoanAndItem;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanRepository;
+import org.folio.circulation.domain.representations.ChangeItemStatusRequest;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.ItemRepository;
+import org.folio.circulation.support.NoContentResult;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.circulation.support.http.server.WebContext;
+
+import java.util.concurrent.CompletableFuture;
+
+public abstract class ChangeStatusResource<T extends ChangeItemStatusRequest> extends Resource {
```

Comment:
Personally, I'd prefer to use composition rather than inheritance for extracting the duplication in these two processes.

Reply:
`ChangeItemStatusService` has been implemented and `ChangeStatusResource` was removed.

Reply:
Oh, thanks